raster-foundry/raster-foundry
# Add QP for ownership filtering

## Overview

This PR does a few things:
- updates babel and eslint to handle newer ES features (like object spread and rest)
- updates the pagination service `updatePageParam` to take a final `extraParams` argument that allows us to easily add the `ownership` QP
- updates the project list controller to pull the `currentOwnershipFilter` value from the QP if present and also uses the updated pagination service method above to set the `ownership` qp anytime it does the fetch

### Checklist

- [NA] Styleguide updated, if necessary
- [NA] [Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated
- [NA] Symlinks from new migrations present or corrected for any new migrations
- [NA] Any content changes are properly templated using `BUILDCONFIG.APP_NAME`
- [NA] Any new SQL strings have tests

### Demo

![image](https://user-images.githubusercontent.com/2442245/49886885-d1ed8d80-fe08-11e8-80b3-bec21401b763.png)

### Notes

It'll be important to test this on staging since the babel and eslint things were changed which could potentially affect the build process

## Testing Instructions

 * run `yarn run build-nginx` and make sure that runs successfully
 * start the front-end (working with staging is fine) and test the ownership filtering

Closes #4373
